File: augmentos_manager/ios/AugmentOS_Manager/RNEventEmitter.m
//
//  RNEventEmitter.m
//  AugmentOS_Manager
//
//  Created by Matthew Fosse on 3/4/25.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(RNEventEmitter, RCTEventEmitter)
  RCT_EXTERN_METHOD(supportedEvents)
@end

File: augmentos_manager/ios/AugmentOS_Manager/RNEventEmitter.swift
//
//  RNEventEmitter.swift
//  AugmentOS_Manager
//
//  Created by Matthew Fosse on 3/4/25.
//

import Foundation
import React

@objc(RNEventEmitter)
open class RNEventEmitter: RCTEventEmitter {

  public static var emitter: RCTEventEmitter!

  override init() {
    super.init()
    RNEventEmitter.emitter = self
  }

  open override func supportedEvents() -> [String] {
    // add more as needed
    ["onReady", "onPending", "onFailure", "onConnectionStateChanged", "CoreMessageIntentEvent"]
  }
}
